### PR TITLE
Market liquidity integration

### PR DESCRIPTION
**Pull request by JohnDanz**
https://github.com/AugurProject/augur/issues/2498

addressed some comments from previous PR, added more styling and confirmed getter is working across markets. WIP for the overall issue, still need to update validation and messaging around tips for users to help with passing the filter and suggestions.

### CODE REVIEW

File: packages/augur-sdk/src/utils.ts [hunk 1]
```diff
@@ -11,6 +11,12 @@ import { toAscii } from './state/utils/utils';
 
 export const QUINTILLION = new BigNumber(10).pow(18);
 
+export function marketNameToType(
+  marketName: string
+): number {
+  return MarketType[marketName];
```
Comment (bthaile) [inline packages/augur-sdk/src/utils.ts]: need to handle error conditions. like name isn't in list. prob just throw an error.

File: packages/augur-ui/src/modules/create-market/components/visibility.tsx [hunk 2]
```diff
@@ -66,65 +69,64 @@ export default class Visibility extends Component<
...
-      case SCALAR:
-        marketTypeNumber = 2;
-        break;
-      default:
-        break;
-    }
-    const formattedOrderBook = {};
-
+    // backend uses YesNo, ui uses yesNo, this ensures capital Y if that's the case.
+    const marketName = marketType.charAt(0).toUpperCase() + marketType.slice(1);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/visibility.tsx]: too much of a hack for my tastes. 
We need to use same as MarketInfos marketType values.

need to fix with our constants to be same as sdk, or when the value is set on NewMarket object when user picks the type of market. 

use sdk values: MarketType.YesNo, ....

File: packages/augur-ui/src/modules/create-market/components/visibility.tsx [hunk 3]
```diff
@@ -140,26 +142,32 @@ export default class Visibility extends Component<
...
+        >
           <LargeSubheaders
             link
             header="Market visibility"
             subheader="To ensure your market is visible to users you must pass the spread filter check. To improve the ranking or visiblity of your market, ensure you add good liquidity to each outcome."
           />
           <SmallSubheadersTooltip
             header="default Spread filter check"
-            subheader="Fail"
+            subheader={hasLiquidity ? 'Pass' : 'Fail'}
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/visibility.tsx]: if Pass and Fail are used in other places create a constant.

**Review by bthaile — CHANGES_REQUESTED** [review]
one real change to be consistent and reduce hacking up strings.

File: packages/augur-ui/src/modules/create-market/components/visibility.tsx [hunk 4]
```diff
@@ -66,70 +119,125 @@ export default class Visibility extends Component<
...
+      minPriceBigNumber,
+      tickSizeBigNumber
+    );
+    const onChainSpread = createBigNumber(`.${MAX_SPREAD_10_PERCENT}`).times(
+      maxOnChain
+    );
+    Object.entries(orderBook).forEach(([outcome, orders]: Array<any>) => {
+      const numOutcome = parseInt(outcome);
+      formattedOrderBook[numOutcome] = { bids: [], asks: [] };
+      let spreadCheck = [createBigNumber('0'), createBigNumber('0')];
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/visibility.tsx]: I'm not following the logic

why is `spreadCheck` an array

**Review by bthaile — COMMENTED** [review]
need to break up logic to tell user about need buys | sells or meet spread before processing the order book and sending to sdk for ranking. 

comments communicated in discord.

File: packages/augur-ui/src/modules/create-market/components/visibility.tsx [hunk 5]
```diff
@@ -46,8 +74,55 @@ export default class Visibility extends Component<
...
+
+    let closestOutcome = 0;
+
+    Object.entries(newMarket.orderBook).forEach(
+      ([outcome, orders]: Array<any>) => {
+        const { asks, bids, spread } = formatOrderBook(orders);
+        if (asks.length > 0) validations.hasSells = true;
+        if (bids.length > 0) validations.hasBuys = true;
+        if (asks.length > 0 || bids.length > 0) {
+          closestOutcome = parseInt(outcome);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/visibility.tsx:102]: outcome should already be a number, should be typed

File: packages/augur-ui/src/modules/create-market/components/common.tsx [hunk 6]
```diff
@@ -131,8 +131,8 @@ export const SmallSubheaders = (props: SubheadersProps) => (
 );
 
 export interface SubheadersTooltipProps {
-  header: string;
-  subheader: string;
+  header: string | Element;
+  subheader: string | Element;
```
Comment (bthaile) [inline packages/augur-ui/src/modules/create-market/components/common.tsx:135]: where is this an Element, didn't see it.

**Review by bthaile — APPROVED** [review]
lgtm, awesome. code looks clean and readable. 👍

File: packages/augur-ui/src/modules/create-market/components/visibility.tsx (hunk 5 above)
Comment (bthaile) [reply packages/augur-ui/src/modules/create-market/components/visibility.tsx:102]: Object.entries converts outcome to string, just a note.

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 💚

 This link will always point to this PR's latest changes: https://peek.run/o9ph6l1

**Pull request merged**